app/src/main/java/ar/scacchipa/twittercloneapp/viewmodel/SplashViewModel.kt

```diff
@@ -4,11 +4,14 @@ import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import ar.scacchipa.twittercloneapp.data.UserAccessTokenDomain
+import ar.scacchipa.twittercloneapp.domain.IRecoveredAccessTokenUseCase
 import ar.scacchipa.twittercloneapp.domain.SplashTimerUseCase
 import kotlinx.coroutines.launch
 
 class SplashViewModel(
-    private val splashTimer: SplashTimerUseCase
+    private val splashTimer: SplashTimerUseCase,
+    private val recoveredAccessTokenUseCase: IRecoveredAccessTokenUseCase
```

Comment:
shouldn't be called the "RecoverAccessTokenUseCase" instead of the implementation?? also the naming can be better

Reply:
what I mean is that you're adding a extra file, you already should do the logic to get the credentials in the UseCase there is no need for an interface in this case

Reply:
The RecoverAccessTokenUseCase could be called RecoverCredentials.

I could create a HasCredentialsUseCase thar return if the credential is stored. What do you think?
(I created the RecoverAccessTokenUseCase thinking in which it is necessary)